ddtrace/internal/agent.py

```diff
@@ -1,16 +1,25 @@
 import json
+from typing import TYPE_CHECKING
 from typing import Union
 
 from ddtrace.internal.logger import get_logger
 from ddtrace.settings._agent import config
 
-from .http import HTTPConnection
-from .http import HTTPSConnection
-from .uds import UDSHTTPConnection
 from .utils.http import get_connection
 
 
-ConnectionType = Union[HTTPSConnection, HTTPConnection, UDSHTTPConnection]
+DEFAULT_HOSTNAME = "localhost"
+DEFAULT_TRACE_PORT = 8126
+DEFAULT_UNIX_TRACE_PATH = "/var/run/datadog/apm.socket"
+DEFAULT_UNIX_DSD_PATH = "/var/run/datadog/dsd.socket"
+DEFAULT_STATS_PORT = 8125
+
+if TYPE_CHECKING:  # pragma: no cover
+    from .http import HTTPConnection
+    from .http import HTTPSConnection
+    from .uds import UDSHTTPConnection
+
+    ConnectionType = Union[HTTPSConnection, HTTPConnection, UDSHTTPConnection]
```

Comment:
```suggestion
ConnectionType = Union["HTTPSConnection", "HTTPConnection", "UDSHTTPConnection"]
```

This might solve some problems, or needs for `if TYPE_CHECKING:` in other modules?

Reply:
Oh brilliant. I didn't realize this was possible. I'll read up more about python type checking and make these updates throughout the PR.

Reply:
Oh brilliant. I didn't realize this was possible. I'll read up more about python type checking and make these updates throughout the PR.